exporter/views.py

```diff
@@ -120,9 +129,22 @@ def get(self, request, *args, **kwargs):
         # Download zip
         with open("%s.zip" % export_base_dir, "rb") as zip_file:
             response = HttpResponse(zip_file, content_type="application/zip")
-        response["Content-Disposition"] = "attachment; filename=%s.zip" % campaign_slug
+
+        export_filename = "%s-%s.zip" % (
+            campaign_slug,
+            datetime.today().isoformat(timespec="minutes"),
+        )
+
+        response["Content-Disposition"] = "attachment; filename=%s" % export_filename
 
         # Upload zip to S3 bucket
+        s3_bucket = getattr(settings, "S3_BUCKET_NAME", None)
+
+        if s3_bucket:
+            s3 = boto3.resource("s3")
+            s3.Bucket(s3_bucket).upload_file(
+                "%s.zip" % export_base_dir, "exporter/%s" % export_filename
+            )
 
         # Clean up temp folders & zipfile once exported
         shutil.rmtree(export_base_dir)
```

Comment:
We need to handle the case where an error prevents this final cleanup step from running. That could either be a try-finally block or using the context-manager form of [`TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) but it perhaps also says that we should move the the core of this into a separate function so it'd basically be something like:

```python
    def get(self, request, *args, **kwargs):
        campaign_slug = self.kwargs["campaign_slug"]
        assets = get_latest_transcription_data(campaign_slug, True)

        with TemporaryDirectory(prefix=campaign_slug) as export_base_dir:
            return do_the_export(…)

    def do_the_export(…):
        for asset in assets:
            …

        …
        
        if s3_bucket:
            s3 = boto3.resource("s3")
            s3.Bucket(s3_bucket).upload_file(
                "%s.zip" % export_base_dir, "exporter/%s" % export_filename
            )

        return response
```